### PR TITLE
refactor: use `cache.get()` for snowflakes, `resolve()` otherwise

### PR DESCRIPTION
**Pull request by sdanialraza**
**Please describe the changes this PR makes and why it should be merged:**
This makes it consistent to use `cache.get()` for snowflakes and `resolve()` for everything else.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Review by didinele — DISMISSED** [review]
surely this won't somehow break anything

File: packages/discord.js/src/managers/ThreadMemberManager.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ class ThreadMemberManager extends CachedManager {
     const memberResolvable = super.resolve(member);
     if (memberResolvable) return memberResolvable;
     const userResolvable = this.client.users.resolveId(member);
-    if (userResolvable) return super.resolve(userResolvable);
+    if (userResolvable) return super.cache.get(userResolvable);
```
Comment (Jiralite) [inline packages/discord.js/src/managers/ThreadMemberManager.js:75]: It looks more appropriate to rename the variable to `userId`.

File: packages/discord.js/src/managers/PresenceManager.js [hunk 2]
```diff
@@ -38,8 +38,8 @@ class PresenceManager extends CachedManager {
   resolve(presence) {
     const presenceResolvable = super.resolve(presence);
     if (presenceResolvable) return presenceResolvable;
-    const UserResolvable = this.client.users.resolveId(presence);
-    return super.resolve(UserResolvable);
+    const userResolvable = this.client.users.resolveId(presence);
+    return super.cache.get(userResolvable) ?? null;
```
Comment (Jiralite) [inline packages/discord.js/src/managers/PresenceManager.js:42]: It looks more appropriate to rename the variable to `userId`.

File: packages/discord.js/src/structures/ThreadMember.js [hunk 3]
```diff
@@ -87,7 +87,7 @@ class ThreadMember extends Base {
    * @readonly
    */
   get user() {
-    return this.client.users.resolve(this.id);
+    return this.client.users.cache.get(this.id);
```
Comment (Jiralite) [inline packages/discord.js/src/structures/ThreadMember.js:90]:
```suggestion
    return this.client.users.cache.get(this.id) ?? null;
```

**Review by Jiralite — CHANGES_REQUESTED** [review]

**Review by Jiralite — APPROVED** [review]

File: packages/discord.js/src/managers/GuildBanManager.js [hunk 4]
```diff
@@ -158,7 +158,7 @@ class GuildBanManager extends CachedManager {
       reason: options.reason,
     });
     if (user instanceof GuildMember) return user;
-    const _user = this.client.users.resolve(id);
+    const _user = this.client.users.cache.get(id) ?? null;
```
Comment (vladfrangu) [inline packages/discord.js/src/managers/GuildBanManager.js:161]: This is one of the cases, like in `packages/discord.js/src/client/actions/ThreadListSync.js`, where you don't need `?? null`

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Comment by DJj123dj** [comment]
This change DID indeed break a lot of bots. It might not look like it at a first glance, but the `cache.get()` method only allows ID's/snowflakes. It returns `undefined` when trying to resolve a `Role` or `User` class in the `PermissionOverwrites`.

The param type still allows the `Role` & `User` class to be used even if they aren't processed and will just return `undefined`.

😄 But no worries, in the latest version of discord.js is this code already overwritten/modified & now it works again.

**Comment by Jiralite** [comment]
Code talks better! What was the issue? I want to understand what you are talking about.

**Comment by DJj123dj** [comment]
> Code talks better! What was the issue? I want to understand what you are talking about.

Yea, the issue was exactly what is described in pull request #10686 as mentioned by @sdanialraza. The fact that it would return `undefined` when using a `discorde.Role` or `discord.User` in the `id` of the `PermissionOverwrites` instead of a `string` (snowflake)

**Comment by Jiralite** [comment]
We had decoded your message by then! Please open an issue instead of commenting on pull requests in the future with a reproducible code sample next time.

**Pull request merged**